slmigrate/mongohandler.py

```diff
@@ -25,39 +26,82 @@ def stop_mongo(proc):
 def capture_migration(service, action, config):
     if action != constants.capture_arg:
         return
-    # mongo_migration_dir = os.path.join(service.migration_dir, "mongo-dump")
-    # mongo_migration_dir = os.path.join(service.migration_dir, "mongo-dump")
     cmd_to_run = constants.mongo_dump + " --port " + str(config[service.name]['Mongo.Port']) + " --db " + config[service.name]['Mongo.Database'] + " --username " + config[service.name]['Mongo.User'] + " --password " + config[service.name]['Mongo.Password'] + " --out " + constants.mongo_migration_dir + " --gzip"
     subprocess.run(cmd_to_run)
 
 
 def restore_migration(service, action, config):
     if action != constants.restore_arg:
         return
-    # mongo_migration_dir = os.path.join(service.migration_dir, "mongo-dump")
     mongo_dump_file = os.path.join(constants.mongo_migration_dir, config[service.name]['Mongo.Database'])
     cmd_to_run = constants.mongo_restore + " --port " + str(config[service.name]['Mongo.Port']) + " --db " + config[service.name]['Mongo.Database'] + " --username " + config[service.name]['Mongo.User'] + " --password " + config[service.name]['Mongo.Password'] + " --gzip " + mongo_dump_file
     subprocess.run(cmd_to_run)
 
 
+def migrate_document(destination_collection, document):
+    try:
+        print("Migrating " + str(document['_id']))
+        destination_collection.insert_one(document)
+    except pyerr.DuplicateKeyError:
+        print("Docuement " + str(document['_id']) + " already exists. Skipping")
+
+
+def identify_metadata_conflict(destination_collection, source_document):
+    destination_query = {'$and': [{'workspace': source_document['workspace']}, {'path': source_document['path']}]}
+    destination_document = destination_collection.find_one(destination_query)
+    if destination_document:
+        return SimpleNamespace(**{'source_id': source_document['_id'], 'destination_id': destination_document['_id']})
+    else:
+        return None
+
+
+def merge_history_document(source_id, destination_id, destination_db):
+    destination_collection = destination_db.get_collection('values')
+    destination_collection.update_one({'metadataId': source_id}, {'$set': {'metadataId': destination_id}})
+
+
+def migrate_metadata_collection(source_db, destination_db):
+    collection_name = 'metadata'
+    source_collection = source_db.get_collection(collection_name)
+    source_collection_iterable = source_collection.find()
+    destination_collection = destination_db.get_collection(collection_name)
+    for document in source_collection_iterable:
+        conflict = identify_metadata_conflict(destination_collection, document)
+        if conflict:
+            print("Conflict Found! " + "source_id=" + str(conflict.source_id) + " destination_id=" + str(conflict.destination_id))
+            merge_history_document(conflict.source_id, conflict.destination_id, destination_db)
+        else:
+            migrate_document(destination_collection, document)
```

Comment:
Consider naming it `source_document`.  Since it wasn't qualified like the rest, I had to think about whether it was source or destination.  
```suggestion
    for source_document in source_collection_iterable:
        conflict = identify_metadata_conflict(destination_collection, source_document)
        if conflict:
            print("Conflict Found! " + "source_id=" + str(conflict.source_id) + " destination_id=" + str(conflict.destination_id))
            merge_history_document(conflict.source_id, conflict.destination_id, destination_db)
        else:
            migrate_document(destination_collection, source_document)
```

Reply:
Good call. Will do.